raystack/frontier/v1beta1/frontier.proto

```diff
@@ -706,6 +706,75 @@ service FrontierService {
+      tags: "Organization";
+      summary: "List org domains";
+      description: "Returns all domains whitelisted for an organization (both pending and verified if no filters are provided for the state). The verified domains allow users email with the org's whitelisted domain to join the organization without invitation.";
+    };
+  }
+
+  rpc AddOrganizationDomain(AddOrganizationDomainRequest) returns (AddOrganizationDomainResponse) {
+    option (google.api.http) = {
+      post: "/v1beta1/organizations/{org_id}/domains",
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      tags: "Organization";
+      summary: "Add org domain";
+      description: "Add a domain to an organization which if verified allows all users of the same domain to be signed up to the organization without invitation. This API generates a verification token for a domain which must be added to your domain's DNS provider as a TXT record should be verified with Frontier VerifyOrgDomain API before it can be used as an Organization's trusted domain to sign up users.";
+    };
+  }
+
+  rpc RemoveOrganizationDomain(RemoveOrganizationDomainRequest) returns (RemoveOrganizationDomainResponse) {
+    option (google.api.http) = {delete: "/v1beta1/organizations/{org_id}/domains/{id}/remove"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      tags: "Organization";
+      summary: "Remove org domain";
+      description: "Remove a domain from the list of an organization's trusted domains list";
+    };
+  }
+
+  rpc GetOrganizationDomain(GetOrganizationDomainRequest) returns (GetOrganizationDomainResponse) {
+    option (google.api.http) = {get: "/v1beta1/organizations/{org_id}/domains/{id}"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      tags: "Organization";
+      summary: "Get org domain";
+      description: "Get a domain from the list of an organization's whitelisted domains. Returns both verified and unverified domains by their ID";
+    };
+  }
+
+  rpc VerifyOrgDomain(VerifyOrgDomainRequest) returns (VerifyOrgDomainResponse) {
+    option (google.api.http) = {
+      post: "/v1beta1/organizations/{org_id}/domains/{id}/verify", 
+      body: "*"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      tags: "Organization";
+      summary: "Verify org domain";
+      description: "Verify a domain for an organization with a verification token generated by Frontier GenerateDomainVerificationToken API. The token must be added to your domain's DNS provider as a TXT record before it can be verified. This API returns the state of the domain (pending/verified) after verification.";
+    };
+  }
+
+  rpc ListOrganizationsByDomain(ListOrganizationsByDomainRequest) returns (ListOrganizationsByDomainResponse) {
+    option (google.api.http) = {get: "/v1beta1/organizations/domains/{name}"};
```

Comment:
I wonder when we will use it directly, what if instead of putting it with domain name, we put it under user?
Like `/users/{id}/organizations` or `/users/self/organizations` and the response could be
```
message GetOrganizationsByUserResponse {
  repeated Organization organizations = 1;
  repeated Organization joinable_via_domain = 2;
}
```